runtime/vm/ClassInitialization.cpp

```diff
@@ -307,6 +316,9 @@ setInitStatus(J9VMThread *currentThread, J9Class *clazz, UDATA status, j9object_
 			STATE_NAME(status)
 	);
 	initializationLock = (j9object_t)VM_ObjectMonitor::enterObjectMonitor(currentThread, initializationLock);
+#if defined(J9VM_OPT_CRIU_SUPPORT)
+	Assert_VM_false(J9_OBJECT_MONITOR_CRIU_SINGLE_THREAD_MODE_THROW == (UDATA)initializationLock);
```

Comment:
would `J9_OBJECT_MONITOR_ENTER_FAILED` be better? it could replace both assertions

Reply:
Yeah, updated.